app/src/main/java/com/codesoom/assignment/DemoHttpHandler.java

```diff
@@ -0,0 +1,169 @@
+        URI requestURI = exchange.getRequestURI();
+        String path = requestURI.getPath();
+        System.out.println(method + " " + path);
+
+        InputStream inputStream = exchange.getRequestBody();
+        String body = new BufferedReader(new InputStreamReader(inputStream))
+                .lines()
+                .collect(Collectors.joining("\n"));
+
+        // path가 아이디를 가지는지 여부
+        boolean hasTaskId = Pattern.matches("/tasks/[0-9]+$", path);
+
+        // 콘텐츠 초기화
+        String content = null;
+
+        // GET /tasks
+        if (method.equals("GET") && path.equals("/tasks")) {
+            content = tasksToJson(tasks);
+        }
+
+        // GET /tasks/{id}
+        if (method.equals("GET") && hasTaskId) {
+            Long id = extractIdFromPath(path);
+            Task findTask = findTaskById(id);
+            content = taskToJson(findTask);
+        }
+
+        // POST /tasks
+        if (method.equals("POST") && path.equals("/tasks")) {
+            Task task = createTaskWithId(body);
+            content = taskToJson(task);
+            tasks.add(task);
+        }
+
+        // Delete /tasks/{id}
+        if (method.equals("DELETE") && hasTaskId) {
+            Long id = extractIdFromPath(path);
+            Task findTask = findTaskById(id);
+            tasks.remove(findTask);
+        }
+
+        // PUT,PATCH /tasks/{id}
+        // 여기서는 Task 객체의 변경 가능한 필드가 title 뿐이므로 put, patch가 동일하게 동작한다.
+        if ((method.equals("PATCH") || method.equals("PUT")) && hasTaskId) {
+            Long id = extractIdFromPath(path);
+            Task findTask = findTaskById(id);
+            Task inputTask = toTask(body);
+            findTask.setTitle(inputTask.getTitle());
+        }
+
+        exchange.sendResponseHeaders(200, content.getBytes().length);
+        OutputStream responseBody = exchange.getResponseBody();
+        responseBody.write(content.getBytes());
+        responseBody.flush();
+        responseBody.close();
+    }
+
+    /**
+     * Task 객체 생성
+     * @param body
+     * @return
+     * @throws JsonProcessingException
+     */
+    private Task createTaskWithId(String body) throws JsonProcessingException {
+        Task task;
+        task = toTask(body);
+        if (tasks.size() == 0) {
+            task.setId(1L);
+        } else {
+            long maxId = tasks.stream().mapToLong(Task::getId).max().orElseThrow(NoSuchElementException::new);
+            task.setId(maxId + 1L);
+        }
+        System.out.println("task = " + task);
+
+        return task;
+    }
+
+    /**
+     * Id로 task 찾기
+     *
+     * @param id 아이디
+     * @return Task
+     */
+    private Task findTaskById(Long id) {
+        Task findTask = tasks.stream()
+                .filter(t -> t.getId().equals(id))
+                .findFirst()
+                .orElseThrow(NoSuchElementException::new);
+        return findTask;
+    }
+
+    /**
+     * Path에서 Id 추출하기
+     *
+     * @param path url 경로
+     * @return Id
+     */
```

Comment:
```suggestion
    /**
     * 주어진 path에서 식별자 숫자를 추출해 리턴합니다.
     *
     * @param path url 경로
     * @return Id 식별자
     */
```

이렇게 수정한 javadoc을 3번 정도 읽어보세요. 어색한 부분이 있나요? 있다면 왜 그렇게 느껴지나요?

Reply:
음... 몇 번 읽어보았지만 제가 생각했을 때는 어색한 부분이 없습니다 .. ㅎㅎㅎ